### ISSUE TITLE
Hellas map bug on the cheap ocean spot

### ISSUE THREAD

**Issue by Crysthorn** [bafolts/terraforming-mars#3098]
In our last TM game we were playing on Hellas and on gen 5 Greens were ruling. My friend Julianna had 20 MC and Wildlife Dome in hand - she played it, spent 15 MC, the game immediately gave her 4 MC from Greens, so she ended up with 9 MC. The problem is that the game didn't let her place the greenery on the cheap ocean spot on the south - she had neighbouring city, she had 6 MC to pay (even a bit more than that), but she had to place the greenery elsewhere.

Game ID was gb99678edf721

![screenshot](https://user-images.githubusercontent.com/77864976/113496466-1b8ea480-94fa-11eb-8e81-887f5b37a5b2.PNG)

**Comment by vsrisuknimit** [comment]
![image](https://user-images.githubusercontent.com/14239220/113599083-1c176000-960c-11eb-8a96-e372217d6ec9.png)

This is also a problem for Tharsis placing a city such that they have 3 mc left. Tharsis should get 3 mc on the city placement and should be allowed to place a city there.

**Comment by Crysthorn** [comment]
Yes, this seems to be the same issue. What's weird is that the game seems to resolve things in the correct order, because Julianna was awarded 4MC from Greens rule immediately and her cash count showed 9MC when she was asked to select space for a greenery. I have no idea why the cheap ocean spot wasn't available - it definitely should be.

**Comment by kberg** [comment]
From the FAQ:

> The 6 MC cost for placing a bonus is an effect that can be resolved any time during
> the action to place a tile there. For example, you can place a tile there, place the
> ocean, and then use MC from that placement to pay for the 6 MC cost.

Just a note: in the game you described, the science payout was too soon: the "When" of the policy forces that that to be granted after the tile is put down. That said, this is the same issue as #1823 and #1754 and won't be resolveable until a good undo is in place.

**Comment by kberg** [comment]
#5899 doesn't fix this, but it does take a step in the right direction for predicting whether someone can afford tile placement in a space that has additional costs..
